libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewApi.kt

```diff
@@ -73,36 +61,115 @@ class NavigationViewApi internal constructor(
      */
-    fun setDestination(point: Point) {
-        store.dispatch(DestinationAction.SetDestination(Destination(point)))
-    }
+    abstract fun setDestination(point: Point)
 
     /**
      * Enables trip session based on real gps updates.
      */
-    fun enableTripSession() {
-        store.dispatch(TripSessionStarterAction.EnableTripSession)
-    }
+    abstract fun enableTripSession()
 
     /**
      * Enables replay trip session based on simulated locations.
      */
-    fun enableReplaySession() {
-        store.dispatch(TripSessionStarterAction.EnableReplayTripSession)
-    }
+    abstract fun enableReplaySession()
 
     /**
      * Checks if the current trip is being simulated.
      */
-    fun isReplayEnabled(): Boolean {
-        return store.state.value.tripSession.isReplayEnabled
-    }
+    abstract fun isReplayEnabled(): Boolean
+
+    /**
+     * Clear Route data and request [NavigationView] to enter Free Drive state.
+     *
+     * [NavigationViewListener.onFreeDrive] or [NavigationViewListener.onDestinationPreview]
+     * (depending if destination has been set) will be called once [NavigationView] enters
+     * Free Drive state.
+     */
+    abstract fun startFreeDrive()
+
+    /**
+     * Request [NavigationView] to enter Route Preview state.
+     *
+     * [NavigationViewListener.onRoutePreview] will be called once [NavigationView] enters
+     * Route Preview state.
+     *
+     * @throws IllegalStateException if either Destination or Preview Routes are not set.
+     */
+    @Throws(IllegalStateException::class)
+    abstract fun startRoutePreview()
+
+    /**
+     * Request [NavigationView] to enter Active Navigation state.
+     *
+     * [NavigationViewListener.onActiveNavigation] will be called once [NavigationView] enters
+     * Active Navigation state.
+     *
+     * @throws IllegalStateException if either Destination or Preview Routes are not set.
+     */
+    @Throws(IllegalStateException::class)
+    abstract fun startNavigation()
+
+    /**
+     * Request [NavigationView] to enter Arrival state.
+     *
+     * [NavigationViewListener.onArrival] will be called once [NavigationView] enters
+     * Arrival state.
+     */
+    abstract fun startArrival()
+
+    /**
+     * Clear Destination and Route data, and update [NavigationView] to Free Drive state.
+     */
+    abstract fun endNavigation()
+}
+
+/**
+ * Sets a [destination], preview [routes] and request [NavigationView] to enter Route Preview state.
+ *
+ * It is a shorthand for calling the following with non-empty [routes].
+ * ```
+ *     setDestination(point)
+ *     setPreviewRoutes(routes)
+ *     startRoutePreview()
+ * ```
+ *
+ * @throws IllegalArgumentException if the [routes] argument is an empty list.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+@Throws(IllegalArgumentException::class)
+fun NavigationViewApi.startRoutePreview(destination: Point, routes: List<NavigationRoute>) {
```

Comment:
Do we need `destination` as an input? We can deduce it internally and set it to the last point from `routes`.